external_resources_io/config.py

```diff
@@ -0,0 +1,31 @@
+from pydantic import field_validator
+from pydantic_settings import BaseSettings
+
+
+class Config(BaseSettings):
+    """Environment Variables."""
+
+    # general settings
+    action: str = "Apply"
+    dry_run: bool = True
+    log_level: str = "INFO"
+
+    # app-interface input related
+    input_file: str = "/inputs/input.json"
+
+    backend_tf_file: str = "module/backend.tf"
+    outputs_file: str = "tmp/outputs.json"
+    plan_file_json: str = "tmp/plan.json"
+    terraform_cmd: str = "terraform"
+    tf_vars_file: str = "module/tfvars.json"
+    variables_tf_file: str = "module/variables.tf"
+
+    @field_validator("action", mode="before")
+    @classmethod
+    def lower_action(cls, v: str) -> str:
+        """Transform the action to lowercase."""
+        return v.lower()
+
+
+def get_env_var_name(field_name: str) -> str:
+    return Config.model_fields[field_name].alias or field_name.upper()
```

Comment:
`alias` is not the only source, https://docs.pydantic.dev/latest/concepts/pydantic_settings/#environment-variable-names

how about we make each ENV names explicit as constant str/enum, then config it as `Field` in `Config`, use constant directly in `cli.py`

Reply:
use `_extract_field_info` private method? And still need to know the field name as string to use `get_env_var_name`. How about extract env name directly like 

```python
VARIABLES_TF_FILE = "VARIABLES_TF_FILE"
```

and use in both places

```python
envvar = VARIABLES_TF_FILE
```

```python
variables_tf_file: str = Field("module/variables.tf", validation_alias=VARIABLES_TF_FILE)
```

Reply:
To be honest, I don't want to specify the same string repeatedly. I would use `alias` instead of `validation_alias`, but mypy doesn't allow variables for the `alias` attribute. An example would be:

```python
ENV_VAR_ACTION = "ACTION"
ENV_VAR_DRY_RUN = "DRY_RUN"
ENV_VAR_VARIABLES_TF_FILE = "VARIABLES_TF_FILE"


class Config(BaseSettings):
    action: Action = Field(Action.APPLY, validation_alias=ENV_VAR_ACTION)
    dry_run: bool = Field(default=True, validation_alias=ENV_VAR_DRY_RUN)
    variables_tf_file: str = Field(
        "module/variables.tf", validation_alias=ENV_VAR_VARIABLES_TF_FILE
    )
```

compared to

```python
class Config(BaseSettings):
    action: Action = Action.APPLY
    dry_run: bool = True
    variables_tf_file: str = "module/variables.tf"
```

Reply:
need pydantic mypy plugin? https://docs.pydantic.dev/latest/integrations/mypy/

what I don't like is the magic value used like `get_env_var_name("variables_tf_file")`, it's easier to maintain if it can be something like `get_env_var_name(Config.variables_tf_file.field_name)` (non exist) or `get_env_var_name(ConfigFields.variables_tf_file)`. This requires a new constant class/enum anyway, and specify value is easier than field name.

Reply:
The pydantic mypy plugin does the trick! I've replaced the `get_env_var_name` method with a class.